templates/secret-generation.yaml

```diff
@@ -0,0 +1,116 @@
+{{- if .Values.worker.enabled  }}
+{{- if .Values.secrets.create }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  annotations:
+    description: |
+      A Role that allows the secret generator below to create secrets for Concourse web and worker nodes.
+  name: secrets-generator
+  namespace: {{ .Release.Namespace }}
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - list
+  - get
+  - create
+  - update
+  - delete
+  - patch
+
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: secrets-generator
+  namespace: {{ .Release.Namespace }}
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  annotations:
+  name: secrets-generator
+  namespace: {{ .Release.Namespace }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: secrets-generator
+subjects:
+- kind: ServiceAccount
+  name: secrets-generator
+  namespace: {{ .Release.Namespace }}
+
+---
+# This Job will patch the web and worker secrets in order to add 
+# key and ssh keys for 
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: secrets-generator-{{ .Release.Revision }}
+  namespace: {{ .Release.Namespace }}
+spec:
+  template:
+    spec:
+      serviceAccountName: secrets-generator
+      restartPolicy: OnFailure
+      initContainers:
+      - name: generator
+        image: concourse/concourse
+        imagePullPolicy: Always
+        command:
+        - /bin/bash
+        - -c
+        - |
+          /usr/local/concourse/bin/concourse generate-key -t rsa -f ./secrets/session-signing-key
+          /usr/local/concourse/bin/concourse generate-key -t ssh -f ./secrets/host-key
+          /usr/local/concourse/bin/concourse generate-key -t ssh -f ./secrets/worker-key
+
+        volumeMounts:
+          - name: secrets
+            mountPath: /secrets
+      containers:
+      - name: secret-creator 
+        image: bitnami/kubectl
```

Comment:
Should we use some other image? We need an image with kubectl in it in order to patch the secrets.